hack/build-and-lint.sh

```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+
+set -e -x -u
+
+./hack/build.sh
+./hack/linter.sh
```

Comment:
I wonder if this should be added to test-all.sh. I rarely build because the test scripts have a build in them, so will likely miss any linting errors until they hit github.

Reply:
The intention in doing so is to keep the test-all script separate in CI. It's nice to run the linting/build and test in parallel and report results separately. 

One thing that we could do is have a script that runs everything (i.e., check.sh) that runs everything to determine whether or not everything is ready to be pushed up. Then the test all script can be run on its own in CI.

Not strongly opinionated though. Am happy to just load everything into one CI job if that's easier too.

Reply:
Alright, that makes sense. I dont think there is a need for another script yet. We can add one in the future if develop starts to get a lot of commits that just fix linting errors.